lang/c++/include/avro/DataFile.hh

```diff
@@ -54,6 +54,11 @@ const int SyncSize = 16;
  */
 typedef std::array<uint8_t, SyncSize> DataFileSync;
 
+/**
+ * The metadata for the data file.
+ */
+typedef std::map<std::string, std::vector<uint8_t>> Metadata;
```

Comment:
For completeness, it'll be great if we copy the documentation for metadata from Avro specification. That is, what is expected of the contents and the predefined keys supported by the spec.

Reply:
Updated. Thanks!